src/nbl/ext/DepthPyramidGenerator/DepthPyramidGenerator.cpp

```diff
@@ -0,0 +1,244 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include <nbl/ext/DepthPyramidGenerator/DepthPyramidGenerator.h>
+
+using namespace nbl;
+using namespace nbl::core;
+using namespace nbl::asset;
+using namespace nbl::video;
+
+namespace nbl
+{
+namespace ext
+{
+namespace DepthPyramidGenerator
+{
+
+DepthPyramidGenerator::DepthPyramidGenerator(IVideoDriver* driver, IAssetManager* am, core::smart_refctd_ptr<IGPUImageView> inputDepthImageView,
+	core::smart_refctd_ptr<IGPUImageView>* outputDepthPyramidMips,
+	const Config& config)
+	: m_driver(driver)
+{
+	// TODO: complete supported formats
+	switch (config.outputFormat)
+	{
+	case EF_R16_SFLOAT:
+	case EF_R32_SFLOAT:
+		break;
+	case EF_R32G32_SFLOAT:
+	case EF_R16G16_SFLOAT:
+		if (config.op != E_MIPMAP_GENERATION_OPERATOR::BOTH)
+			assert(false);
+		break;
+	default:
+		assert(false);
+	}
+
+
+	core::smart_refctd_ptr<IGPUDescriptorSetLayout> dsLayout;
+	{
+		IGPUSampler::SParams params;
+		params.TextureWrapU = ISampler::ETC_MIRROR;
+		params.TextureWrapV = ISampler::ETC_MIRROR;
+		params.TextureWrapW = ISampler::ETC_MIRROR;
+		params.BorderColor = ISampler::ETBC_FLOAT_OPAQUE_BLACK;
```

Comment:
you need to pad with:
- 0.f reduction operator is MAX
- 1.f reduction operator is MIN  

But if you're doing BOTH, then you need to pad with 1.f in red and 0.f in green.

Obviously the enums are really limited there, so I propose you apply a SComponentMapping (in the image view used for sampling) such that Green goes to Alpha (yes this does necessitate creating extra image views)